src/app/shell/details/achievements/achievements.component.ts

```diff
@@ -13,30 +13,39 @@ import { UserState } from 'src/app/shared/store/user.state';
   templateUrl: './achievements.component.html',
   styleUrls: ['./achievements.component.scss'],
 })
-export class AchievementsComponent implements OnInit {
-  readonly noResultAchievements = NoResultsTitle.noAchievements;
-  @Input() achievements: Achievement[];
-
+export class AchievementsComponent implements OnInit {  
   @Select(UserState.achievements)
   achievements$: Observable<Achievement[]>;
-  @Input() workshop: Workshop;
+  @Select(UserState.selectedWorkshop) workshop$: Observable<Workshop>;
+
+  @Input() achievements: Achievement[];
+  workshop: Workshop;
+  readonly noResultAchievements = NoResultsTitle.noAchievements;
   destroy$: Subject<boolean> = new Subject<boolean>();
 
   constructor(private store: Store) {}
 
   ngOnInit(): void {   
-    this.getAchievements();    
     this.achievements$
       .pipe(
         takeUntil(this.destroy$),
         filter((achievements) => !!achievements)
       ).subscribe((achievements) => {
         this.achievements = achievements;
       });
+
+    this.workshop$
```

Comment:
I think this is redundant, you can remove at all this subscription, because you can receive workshop as input from the details